src/ui/AccessibleLinksFormatter.cpp

```diff
@@ -0,0 +1,170 @@
+/*!********************************************************************
+
+ Audacity: A Digital Audio Editor
+
+ @file AccessibleLinksFormatter.h
+ @brief Define a helper class to format text with link in a way, accessible to VI users.
+
+ Dmitry Vedenko
+ **********************************************************************/
+
+#include "AccessibleLinksFormatter.h"
+
+#include "ShuttleGui.h"
+
+#include <unordered_map>
+#include <algorithm>
+
+#include <wx/hyperlink.h>
+
+
+namespace
+{    
+size_t OffsetPosition(size_t position, size_t length)
+{
+   if (position == wxString::npos)
+      return wxString::npos;
+
+   return position + length;
+}
+}
+
+AccessibleLinksFormatter::AccessibleLinksFormatter(TranslatableString message)
+    : mMessage(std::move(message))
+{
+}
+
+AccessibleLinksFormatter& AccessibleLinksFormatter::FormatLink(
+   wxString placeholder, TranslatableString value, std::string targetURL)
+{
+   mFormatArguments.push_back({ 
+       std::move(placeholder), 
+       std::move(value), 
+       {}, 
+       std::move(targetURL)
+   });
+
+   return *this;
+}
+
+AccessibleLinksFormatter& AccessibleLinksFormatter::FormatLink(
+   wxString placeholder, TranslatableString value,
+   LinkClickedHandler handler)
+{
+   mFormatArguments.push_back({ 
+       std::move(placeholder), 
+       std::move(value), 
+       std::move(handler), 
+       {} 
+   });
+
+   return *this;
+}
+
+void AccessibleLinksFormatter::Populate(ShuttleGui& S)
```

Comment:
I think this function and the next would be nicer as const.  The format arguments array should remain unchanged.  This is not imporant enough to block the PR.

Reply:
This would make the algorithm much more complex or use extra memory. The second is feasible, I think

Reply:
It will also allow replacing stable_sort with sort